Feature/floating toolbar

timogi opened the pull request.

Commit:
floating elements

Commit:
fix sidenav and drawing

Commit:
fix map

Commit:
add menu to floating icon list

Commit:
style floating ui

Commit:
sidebar for small devices

Commit:
resolve merge conflict

Commit:
fix linting

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in dd6be2c according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/331

timogi committed:
Merge branch 'dev' into feature/floating-toolbar

Commit:
fix antipatern by deleting non-needed function

Commit:
Merge branch 'feature/floating-toolbar' of github.com:zskarte/zskarte-client into feature/floating-toolbar

Commit:
fix linting

pascalknecht commented on src/app/coordinates/coordinates.component.scss:
```diff
@@ -0,0 +1,8 @@
+  background: white;
+  display: flex;
+  flex-direction: column;
+  font-size: 0.85em;
```
rem brauchen

pascalknecht commented on src/app/coordinates/coordinates.component.ts:
```diff
@@ -0,0 +1,37 @@
+})
+export class CoordinatesComponent {
+  coordinates: string[] = [];
+  constructor(public state: ZsMapStateService) {
```
private machen

pascalknecht commented on src/app/coordinates/coordinates.component.ts:
```diff
@@ -0,0 +1,37 @@
+    });
+  }
+
+  coordinatesToString(coordinates: Coordinate, format: string): string {
```
format besserer Type machen. String ist zu breit

pascalknecht replied:
Methode auslagern in helpers

pascalknecht commented on src/app/coordinates/coordinates.component.ts:
```diff
@@ -0,0 +1,37 @@
+    return '';
+  }
+
+  static transformToProjection(coordinates: Coordinate, projection_index: number) {
```
camelCase für projection_index

pascalknecht commented on src/app/sidebar/sidebar-menu/sidebar-menu.component.scss:
Entfernen

pascalknecht commented on src/app/floating-ui/floating-ui.component.scss:
```diff
@@ -0,0 +1,109 @@
+  width: 100%;
+}
+
+button {
```
pointer-events würde ich mit > * wieder aktivieren

pascalknecht commented on src/app/floating-ui/floating-ui.component.scss:
```diff
@@ -0,0 +1,109 @@
+  pointer-events: none;
+}
+
+.main {
```
main class ist doppelt gemoppelt

pascalknecht commented on src/app/floating-ui/floating-ui.component.scss:
```diff
@@ -0,0 +1,109 @@
+  flex-direction: column;
+  align-items: center;
+  justify-content: space-between;
+  gap: 1em;
```
rem verwenden

pascalknecht commented on src/app/floating-ui/floating-ui.component.ts:
```diff
@@ -0,0 +1,82 @@
+  public connectionCount = new BehaviorSubject<number>(0);
+  public isOnline = new BehaviorSubject<boolean>(true);
+  public isReadOnly = new BehaviorSubject<boolean>(false);
+  private _deviceTrackingLayer!: VectorLayer<VectorSource>;
```
wird nicht verwendet

pascalknecht committed:
Fixed floating ui

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in ad701ab according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/331

pascalknecht committed:
Implemented feedback

pascalknecht committed:
Merge branch 'feature/floating-toolbar' of github.com:zskarte/zskarte-v3 into feature/floating-toolbar

dfseifert commented on src/app/sidebar/sidebar-menu/sidebar-menu.component.html:
```diff
@@ -0,0 +1,86 @@
+  </button>
+  <button mat-menu-item (click)="showRevokeShareDialog()">
+    <mat-icon aria-hidden="false" [attr.aria-label]="i18n.get('rewokeAccess')">locked</mat-icon>
+    {{ i18n.get('rewokeAccess') }}
```
```suggestion
    {{ i18n.get('revokeAccess') }}
```

dfseifert commented on src/app/sidebar/sidebar-menu/sidebar-menu.component.html:
```diff
@@ -0,0 +1,86 @@
+    {{ i18n.get('generateSingleUseShareLink') }} (readonly)
+  </button>
+  <button mat-menu-item (click)="showRevokeShareDialog()">
+    <mat-icon aria-hidden="false" [attr.aria-label]="i18n.get('rewokeAccess')">locked</mat-icon>
```
```suggestion
    <mat-icon aria-hidden="false" [attr.aria-label]="i18n.get('revokeAccess')">locked</mat-icon>
```

pascalknecht committed:
Removed woke

dfseifert commented on src/app/sidebar/sidebar-menu/sidebar-menu.component.ts:
```diff
@@ -64,8 +64,8 @@ export class ToolbarComponent implements OnDestroy {
 
   isInitialLaunch(): boolean {
```
```suggestion
  // skipcq:  JS-0105
  isInitialLaunch(): boolean {
```

dfseifert commented on src/app/coordinates/coordinates.component.ts:
```diff
@@ -0,0 +1,37 @@
+    });
+  }
+
+  coordinatesToString(coordinates: Coordinate, format: string): string {
```
```suggestion
  // skipcq:  JS-0105
  coordinatesToString(coordinates: Coordinate, format: string): string {
```

pascalknecht committed:
Update src/app/coordinates/coordinates.component.ts

Co-authored-by: Daniel Seifert <github.0gy60@passmail.net>

pascalknecht committed:
Update src/app/sidebar/sidebar-menu/sidebar-menu.component.ts

Co-authored-by: Daniel Seifert <github.0gy60@passmail.net>

pascalknecht committed:
Fixed playwright test

pascalknecht committed:
Merging

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 51870c1 according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/331
